update readme (#1)
KseniyaKuzmina/how-to-colorize-map-contours-loaded-from-shapefiles-using-the-choropleth-colorizer-e4691: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E4691)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# Map for WinForms - Create a Choropleth Map Based on Shapes Loaded From a Shapefile

This example demonstrates how to fill map shape contours with a specific color depending on GDP data from a shapefile (_Countries.dbf_, _Countries.shp_). To load data from a shapefile, use the [ShapefileDataAdapter](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ShapefileDataAdapter?p=netframework).

![Resulting map](Images/resulting-map.png)

To paint the map as in the image above, assign a [ChoroplethColorizer](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ChoroplethColorizer?p=netframework) object to the [VectorItemsLayer.Colorizer](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.VectorItemsLayer.Colorizer) property.

> Then, you need to specify an attribute from a shapefile used to define the shape color. To do this, create a [ShapeAttributeValueProvider](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ShapeAttributeValueProvider) object and specify its [AttributeName](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ShapeAttributeValueProvider.AttributeName) property. Then, assign the provider to the [ChoroplethColorizer.ValueProvider](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ChoroplethColorizer.ValueProvider) property.

Reviewer: ```suggestion
Then, specify an attribute from a shapefile used to define the shape color. To do this, create a [ShapeAttributeValueProvider](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ShapeAttributeValueProvider) object and specify its [AttributeName](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ShapeAttributeValueProvider.AttributeName) property. Then, assign the provider to the [ChoroplethColorizer.ValueProvider](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ChoroplethColorizer.ValueProvider) property.
```

> Split attribute values into ranges that define boundaries for colors. For this, populate the [ChoroplethColorizer.RangeStops](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ChoroplethColorizer.RangeStops) collection with numeric range stops.

Reviewer: ```suggestion
Split attribute values into ranges that define color boundaries. To do this, populate the [ChoroplethColorizer.RangeStops](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ChoroplethColorizer.RangeStops) collection with numeric range stops.
```

Populate the [MapColorizer.ColorItems](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ChoroplethColorizer.ColorItems) colors with items that define colors used to paint shapes. The colorizer automatically associates each color with the specified data range to colorize map shapes.

> If you wish to display information on what each color means, create a [Color Scale legend](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ColorScaleLegend).

Reviewer: ```suggestion
If you want to display information on what each color means, create a [Color Scale legend](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.ColorScaleLegend).
```
